EXOSIMS/Observatory/TwoStarShades_mission.py

```diff
@@ -0,0 +1,567 @@
+from EXOSIMS.Observatory.SotoStarshade import SotoStarshade
+import numpy as np
+import astropy.units as u
+import astropy.constants as const
+import scipy.optimize as optimize
+
+
+class TwoStarShades_mission(SotoStarshade):
+    def __init__(
+        self,
+        scMass=[6000.0, 6000.0],
+        dryMass=[3400.0, 3400.0],
+        counter_1=0,
+        counter_2=0,
+        counter_3=0,
+        counter=0,
+        thrust=450,
+        **specs
+    ):
+        self.counter = counter
+        self.counter_1 = counter_1
+        self.counter_2 = counter_2
+        self.counter_3 = counter_3
+        SotoStarshade.__init__(self, **specs)
+
+        # occulter slew thrust (mN)
+        self.thrust = float(thrust) * u.mN
+
+        # occulters' initial wet mass (kg)
+        self.scMass = (np.array([scMass])) * u.kg
+        # print(self.scMass)
+        # occulters' dry mass(kg)
+        self.dryMass = np.array([dryMass]) * u.kg
+
+        # adding to outspec (Necessary for running ensemble)
+        self._outspec["scMass"] = scMass
+        self._outspec["dryMass"] = dryMass
+        # Acceleration
+        self.ao = self.thrust / self.scMass
+        # acceleration is an array
+        np.array([self.ao])
+        # instantiating fake star catalog, used to generate good dVmap
+
+    def distForces(self, TL, sInd, currentTime):
+        """Finds lateral and axial disturbance forces on an occulter
+
+        Args:
+            TL (:ref:`TargetList`):
+                TargetList class object
+            sInd (int):
+                Integer index of the star of interest
+            currentTime (~astropy.time.Time):
+                Current absolute mission time in MJD
+
+        Returns:
+            tuple:
+                :obj:`~astropy.units.Quantity`:
+                    dF_lateral: Lateral disturbance force in units of N
+                :obj:`~astropy.units.Quantity`:
+                    dF_axial: Axial disturbance force in units of N
+
+        """
+        if self.counter == 0:
```

Comment:
there's way too much duplicated code in this method.  You don't need an if/else structure - just use self.counter as the index.  That is, everywhere where you have something like `self.scMass[:, 0]` you should instead be doing `-self.scMass[:, self.counter]`.  Also, why are you indexing a scMass this way?  I thought it only had 2 elements?

Reply:
fixed